lib/SILOptimizer/Mandatory/TFUtilities.cpp

```diff
@@ -1567,12 +1566,14 @@ tf::createConstTensor(Type elementType, SymbolicValue scalars,
     context.getIdentifier(std::string("value") + tensorSuffix), scalars
   });
 
-  // Add the value$shape attribute.
-  auto shapeSuffix =
-  SILTensorOpInfo::getOperandClassSuffix(SILTensorOpInfo::OperandClass::Shape);
-  attributes.push_back({
-    context.getIdentifier(std::string("value") + shapeSuffix), shape
-  });
+  // Add the value$shape attribute if we have an array value.
+  if (scalars.getKind() == SymbolicValue::Array) {
+    auto shapeId = SILTensorOpInfo::OperandClass::Shape;
+    auto shapeSuffix = SILTensorOpInfo::getOperandClassSuffix(shapeId);
+    attributes.push_back({
+      context.getIdentifier(std::string("value") + shapeSuffix), shape
```

Comment:
given the prev attr is called `value$tensor`, do we want to call this something like `__shape$shape` (i'm suggesting `__` because this attr is a "pseudo attr" -- it won't be lowered to an op attr in the GraphDef)?

Also, do we still need to add things like $tensor, $shape at this point? IIUC, we'd like to delete TF graph lowering code that relies on such specifiers.

Reply:
Yeah, this is a direction I'd love to see!  We'll need some of the specifiers, but not nearly all the ones we have today.